src/SprykerEco/Zed/Payone/Business/Payment/MethodSender/PayonePartialCaptureMethodSender.php

```diff
@@ -0,0 +1,322 @@
+        $this->orderPriceDistributor = $orderPriceDistributor;
+        $this->standartParameterMapper = $standartParameterMapper;
+        $this->shipmentMapper = $shipmentMapper;
+        $this->orderHandlingMapper = $orderHandlingMapper;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayonePartialOperationRequestTransfer $payonePartialOperationRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\CaptureResponseTransfer
+     */
+    public function executePartialCapture(
+        PayonePartialOperationRequestTransfer $payonePartialOperationRequestTransfer
+    ): CaptureResponseTransfer {
+        $distributedPriceOrderTransfer = $this->orderPriceDistributor->distributeOrderPrice(
+            $payonePartialOperationRequestTransfer->getOrder()
+        );
+        $payonePartialOperationRequestTransfer->setOrder($distributedPriceOrderTransfer);
+
+        $paymentEntity = $this->getPaymentEntity($payonePartialOperationRequestTransfer->getOrder()->getIdSalesOrder());
+        $paymentMethodMapper = $this->getPaymentMethodMapper($paymentEntity);
+
+        $requestContainer = $paymentMethodMapper->mapPaymentToCapture($paymentEntity);
+        $requestContainer = $this->preparePartialCaptureOrderItems($payonePartialOperationRequestTransfer, $requestContainer);
+        $captureAmount = $this->calculatePartialCaptureItemsAmount($payonePartialOperationRequestTransfer);
+
+        $captureAmount += $this->getDeliveryCosts($payonePartialOperationRequestTransfer->getOrder());
+        $requestContainer = $this->shipmentMapper->prepareShipment($payonePartialOperationRequestTransfer->getOrder(), $requestContainer);
+
+        $captureAmount += $this->calculateExpensesCost($payonePartialOperationRequestTransfer->getOrder());
+        /** @var \SprykerEco\Zed\Payone\Business\Api\Request\Container\CaptureContainer $requestContainer */
+        $requestContainer = $this->orderHandlingMapper->prepareOrderHandling($payonePartialOperationRequestTransfer->getOrder(), $requestContainer);
+
+        $businessContainer = new BusinessContainer();
+        $businessContainer->setSettleAccount(PayoneApiConstants::SETTLE_ACCOUNT_YES);
+        $requestContainer->setBusiness($businessContainer);
+
+        $requestContainer->setAmount($captureAmount);
+        $this->standartParameterMapper->setStandardParameter($requestContainer, $this->standardParameter);
+
+        $apiLogEntity = $this->initializeApiLog($paymentEntity, $requestContainer);
+
+        $rawResponse = $this->executionAdapter->sendRequest($requestContainer);
+        $responseContainer = new CaptureResponseContainer($rawResponse);
+
+        $this->updateApiLogAfterCapture($apiLogEntity, $responseContainer);
+        $this->updatePaymentPayoneOrderItemsWithStatus(
+            $payonePartialOperationRequestTransfer,
+            $this->getPartialCaptureStatus($responseContainer)
+        );
+
+        $responseMapper = new CaptureResponseMapper();
```

Comment:
* Extract to factory and use as normal dependency.

Reply:
++